src/openeo_gfmap/manager/job_manager.py

```diff
@@ -16,28 +18,62 @@
 from openeo_gfmap.stac import constants
 
 # Lock to use when writing to the STAC collection
-_stac_lock = threading.Lock()
+_stac_lock = Lock()
+
+
+def retry_on_exception(max_retries: int, delay_s: float = 180.0):
```

Comment:
I would set the delay to int

Reply:
Why? It's being transmitted to the `time.sleep(...)` function which accepts a float parameter

Reply:
What is the actual time unit we can expect the retry to be successful on?

Does it make sense to allow ms precission retries?

Reply:
I can't say for sure, this is dependent on issues in the backend. 3 minutes are a bit arbitrary but it's going to be very tricky to collect actual data for an objective decision on this parameter